From 4928a68a1feeb6b7501d8696fb821d245b69151c Mon Sep 17 00:00:00 2001
From: Dinh Nguyen <r00091@freescale.com>
Date: Thu, 14 May 2009 07:51:28 -0500
Subject: [PATCH] ENGR00112269 MX51: Warning is reported when rmmod g_ether

Before unbinding a composite device such as Gadget Serial or Ether,
the driver must disconnect the device before it can be unbinded.

Signed-off-by: Dinh Nguyen <Dinh.Nguyen@freescale.com>
---
 drivers/usb/gadget/arcotg_udc.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/drivers/usb/gadget/arcotg_udc.c b/drivers/usb/gadget/arcotg_udc.c
index 0ea3d00..a376acd 100644
--- a/drivers/usb/gadget/arcotg_udc.c
+++ b/drivers/usb/gadget/arcotg_udc.c
@@ -2202,6 +2202,9 @@ int usb_gadget_unregister_driver(struct usb_gadget_driver *driver)
 		nuke(loop_ep, -ESHUTDOWN);
 	spin_unlock_irqrestore(&udc_controller->lock, flags);
 
+	/* disconnect gadget before unbinding */
+	driver->disconnect(&udc_controller->gadget);
+
 	/* unbind gadget and unhook driver. */
 	driver->unbind(&udc_controller->gadget);
 	udc_controller->gadget.dev.driver = 0;
-- 
1.5.4.4

